### DISCUSSION TITLE
Switch Network

### DISCUSSION THREAD

**Discussion by damianlluch** [NoahZinsmeister/web3-react#434] [Q&A]
Hello, I would like to know how I can do with this library to make a network switch that I need the user to connect to.

Thanks

**Comment by NoahZinsmeister** [comment]
hey @damianlluch , this depends on your connector - but the good news is that the complexity is mostly abstracted for you. typically, if a connector's `activate` method takes something like a chainId parameter (see e.g. https://github.com/NoahZinsmeister/web3-react/blob/main/packages/metamask/src/index.ts#L109), the connector will handle the relevant network switching logic (in this case, using `wallet_switchEthereumChain`)

**Comment by maleekcodes** [comment]
Hey @NoahZinsmeister I tried this and it's pretty good but whenever you are connecting to a completely different chain for example from Ethereum to Polygon, metamask disconnects wallet and ask to try again. Ofcourse you can just call the activate function from the injector again but I am not sure if that's a good practice. Any solution to this?

**Reply by jxiaox** [reply]
I have the same problem

**Comment by nsnull0** [comment]
sorry , i am new to this, but how to use this call on `useWeb3React()`?
connector is extends from AbstractConnector and no related to `activate` with the param of `AddEthereumChainParameter `...

Also i found `MetaMask` class ,, but how to use it with `useWeb3React()`?
is it return from connector from `useWeb3React()`? (i tried it but its not, its AbstractConnector)

Confusing... can someone explain or give me link how to implement switch network?
All i can see from web is only using directly metamask event, not with web3-react.
